google/resource_compute_instance_group.go

```diff
@@ -94,7 +96,104 @@ func resourceComputeInstanceGroup() *schema.Resource {
+	//		will be destroyed and recreated.  Fortunately, in this case, the
+	//		upstream logic will show that there is a diff.  This will be a
+	//		"Computed" diff - as if we had called diff.SetComputed("instances"),
+	//		and that's a good response.
+	// 3) it will be called on an updated resource at "apply" time - that is,
+	//		right when we're about to do something with this resource.  That
+	//		is designed to check whether there really is a diff on the Computed
+	//		field "instances".  Here, we have to get tricky.  We need to show
+	//		a diff, and it can't be a Computed diff (`apply` skips `Update` if
+	//		the only diffs are Computed).  It also can't be a ForceNew diff,
+	//		because Terraform crashes if there's a ForceNew diff at apply time
+	//		after not seeing one at plan time.  We're in a pickle - the Terraform
+	//		state matches our desired state, but is *wrong*.  We add a fake item
+	//		to the "instances" set, so that Terraform sees a diff between the
+	//		state and the desired state.
+
+	oldI, newI := diff.GetChange("instances")
+	oldInstanceSet := oldI.(*schema.Set)
+	newInstanceSet := newI.(*schema.Set)
+	oldInstances := convertStringArr(oldInstanceSet.List())
+	newInstances := convertStringArr(newInstanceSet.List())
+
+	log.Printf("[DEBUG] InstanceGroup CustomizeDiff old: %#v, new: %#v", oldInstances, newInstances)
+	var memberUrls []string
+	config := meta.(*Config)
+	// We can't use getProject() or getZone(), because we only have a schema.ResourceDiff,
+	// not a schema.ResourceData.  We'll have to emulate them like this.
+	project := diff.Get("project").(string)
+	if project == "" {
+		project = config.Project
+	}
+	zone := diff.Get("zone").(string)
+	if zone == "" {
+		project = config.Zone
+	}
+
+	// We need to see what instances are present in the instance group.  There are a few
+	// possible results.
+	// 1) The instance group doesn't exist.  We don't change the diff in this case -
+	//		if the instance is being created, that's the right thing to do.
+	// 2) The instance group exists, and the GCP state matches the terraform state.  In this
+	//		case, we should do nothing.
+	// 3) The instance group exists, and the GCP state does not match the terraform state.
+	//		In this case, we add the string "FORCE_UPDATE" to the list of instances, to convince
+	//		Terraform to execute an update even though there's no diff between the terraform
+	//		state and the desired state.
+	members, err := config.clientCompute.InstanceGroups.ListInstances(
+		project, zone, diff.Get("name").(string), &compute.InstanceGroupsListInstancesRequest{
+			InstanceState: "ALL",
+		}).Do()
+	if err != nil {
+		if gerr, ok := err.(*googleapi.Error); ok && gerr.Code == 404 {
+			// This is where we'll end up at either plan time or apply time on first creation.
+			return nil
+		} else {
+			// Any other errors return them
+			return fmt.Errorf("Error reading InstanceGroup Members: %s", err)
+		}
+	} else {
```

Comment:
nit: this else is unnecessary since the if statement it's attached to will always return